.devcontainer/Dockerfile

```
@@ -0,0 +1,84 @@
+FROM mcr.microsoft.com/vscode/devcontainers/base:0-buster
+
+# [Option] Install zsh
+ARG INSTALL_ZSH="false"
+# [Option] Upgrade OS packages to their latest versions
+ARG UPGRADE_PACKAGES="false"
+# [Option] Enable non-root Docker access in container
+ARG ENABLE_NONROOT_DOCKER="true"
+# [Option] Use the OSS Moby Engine instead of the licensed Docker Engine
+ARG USE_MOBY="true"
+# [Option] Engine/CLI Version
+ARG DOCKER_VERSION="latest"
+
+# Enable new "BUILDKIT" mode for Docker CLI
+ENV DOCKER_BUILDKIT=1
+# debian needs DEV_CONTAINERS_DIR
+ENV DEV_CONTAINERS_DIR=/usr/local/etc/vscode-dev-containers/ 
+
+# Install needed packages and setup non-root user. Use a separate RUN statement to add your
+# own dependencies. A user of "automatic" attempts to reuse an user ID if one already exists.
+ARG USERNAME=automatic
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+COPY library-scripts/*.sh /tmp/library-scripts/
+COPY init.sh /var/init.sh
+COPY first-run-notice.txt /usr/local/etc/vscode-dev-containers/
+
+# Download Kind Docker image so we can bake it into this Docker image instead of having to download it on startup
+RUN bash /tmp/library-scripts/download-frozen-image-v2.sh /var/kindest-node-image kindest/node:v1.24.0
+RUN apt-get update \
+    && /bin/bash /tmp/library-scripts/common-debian.sh "${INSTALL_ZSH}" "${USERNAME}" "${USER_UID}" "${USER_GID}" "${UPGRADE_PACKAGES}" "true" "true" \
+    # Use Docker script from script library to set things up
+    && /bin/bash /tmp/library-scripts/docker-in-docker-debian.sh "${ENABLE_NONROOT_DOCKER}" "${USERNAME}" "${USE_MOBY}" "${DOCKER_VERSION}" \
+    # Clean up
+    && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/* /tmp/library-scripts/ \
+    # Install Make
+    && apt update && apt install build-essential -y && apt install make \
+    # Kubectl autocomplete
+    && sudo apt install bash-completion
+
+### ======= ARC dependencies =======
+# Go
+RUN cd /tmp && wget https://go.dev/dl/go1.19.linux-amd64.tar.gz && tar -C /usr/local -xzf go1.19.linux-amd64.tar.gz && rm go1.19.linux-amd64.tar.gz
+ENV PATH="${PATH}:/usr/local/go/bin"
+
+# VS Code Go extension dependencies.
+RUN go install github.com/cweill/gotests/gotests@latest \
+&& go install github.com/fatih/gomodifytags@latest \
+&& go install github.com/josharian/impl@latest \
+&& go install github.com/haya14busa/goplay/cmd/goplay@latest \
+&& go install github.com/go-delve/delve/cmd/dlv@latest \
+&& go install honnef.co/go/tools/cmd/staticcheck@latest \
+&& go install golang.org/x/tools/gopls@latest
+
+# Kind
+RUN go install sigs.k8s.io/kind@latest
+# Hardcoded '/root/go/bin/' because ${HOME} resolves to '/root/.local/go/bin'
+ENV PATH="${PATH}:/root/go/bin"
```

```suggestion
ENV PATH="${PATH}:${GOPATH}/bin"
```

nit: `go install` by default puts binaries inside `$GOPATH/bin` (unless a `$GOBIN` is set), in this case `/root/go/bin` (this is just to avoid hardcoding)

---

At this point GOPATH will not be visible as env variable. You can see the env with `go env GOPATH` so we can do:
```suggestion
# add $GOPATH/bin to $PATH
RUN echo "PATH=$PATH:$(go env GOPATH)/bin" >> ~/.bashrc
```